diff -r 67d8f41d51d7 dmenu.c
--- a/dmenu.c	Fri Jul 02 06:49:05 2010 +0100
+++ b/dmenu.c	Tue Jul 06 18:38:42 2010 -0500
@@ -254,6 +254,23 @@
 			text[++i] = '\0';
 			match();
 			break;
+		case XK_p:
+			{
+				FILE *fp;
+				char *c;
+				if(!(fp = (FILE*)popen("sselp", "r")))
+					fprintf(stderr, "dmenu: Could not popen sselp\n");
+				c = fgets(text + len, sizeof(text) - len, fp);
+				pclose(fp);
+				if(c == NULL)
+					return;
+			}
+			len = strlen(text);
+			if(len && text[len-1] == '\n')
+				text[--len] = '\0';
+			match();
+			drawbar();
+			return;
 		}
 	}
 	switch(ksym) {
