diff -r ccb2b92086bf dmenu.1
--- a/dmenu.1	Thu Aug 12 15:35:51 2010 +0100
+++ b/dmenu.1	Wed Aug 18 10:59:14 2010 +0100
@@ -5,6 +5,7 @@
 .B dmenu
 .RB [ \-b ]
 .RB [ \-i ]
+.RB [ \-r ]
 .RB [ \-l
 .IR lines ]
 .RB [ \-p
@@ -48,6 +49,9 @@
 .B \-i
 dmenu matches menu items case insensitively.
 .TP
+.B \-r
+dmenu outputs text each time a key is pressed.
+.TP
 .BI \-l " lines"
 dmenu lists items vertically, with the given number of lines.
 .TP
diff -r ccb2b92086bf dmenu.c
--- a/dmenu.c	Thu Aug 12 15:35:51 2010 +0100
+++ b/dmenu.c	Wed Aug 18 10:59:14 2010 +0100
@@ -53,7 +53,7 @@
 static unsigned long normcol[ColLast];
 static unsigned long selcol[ColLast];
 static Atom utf8;
-static Bool topbar = True;
+static Bool topbar = True, incremental = False;
 static DC *dc;
 static Item *items = NULL;
 static Item *matches, *sel;
@@ -317,6 +317,10 @@
 		match();
 		break;
 	}
+	if(incremental) {
+		fprintf(stdout, "%s\n", text);
+		fflush(stdout);
+	}
 	drawmenu();
 }

@@ -487,7 +491,7 @@

 void
 usage(void) {
-	fputs("usage: dmenu [-b] [-i] [-l lines] [-p prompt] [-fn font] [-nb color]\n"
+	fputs("usage: dmenu [-b] [-r] [-i] [-l lines] [-p prompt] [-fn font] [-nb color]\n"
 	      "             [-nf color] [-sb color] [-sf color] [-v]\n", stderr);
 	exit(EXIT_FAILURE);
 }
@@ -505,6 +509,8 @@
 		}
 		else if(!strcmp(argv[i], "-b"))
 			topbar = False;
+		else if(!strcmp(argv[i], "-r"))
+			incremental = True;
 		else if(!strcmp(argv[i], "-i"))
 			fstrncmp = strncasecmp;
 		else if(i == argc-1)
