diff -r f48e2b63129e dmenu.c
--- a/dmenu.c	Tue Oct 27 14:38:03 2009 -0700
+++ b/dmenu.c	Thu Oct 29 13:14:21 2009 -0700
@@ -389,6 +389,23 @@
 		case XK_G:
 			ksym = XK_End;
 			break;
+		case XK_p:
+			{
+				FILE *fp;
+				char *c;
+				if(!(fp = (FILE*)popen("sselp", "r")))
+					fprintf(stderr, "dmenu: Could not popen sselp\n");
+				c = fgets(text + len, sizeof(text) - len, fp);
+				pclose(fp);
+				if(c == NULL)
+					return;
+			}
+			len = strlen(text);
+			if(len && text[len-1] == '\n')
+				text[--len] = '\0';
+			match(text);
+			drawmenu();
+			return;
 		}
 	}
 	switch(ksym) {
