From 0f4694a7feef3b645a257ab02f631f75dc217b08 Mon Sep 17 00:00:00 2001
From: Klemens Nanni <kl3@posteo.org>
Date: Fri, 29 Jul 2016 02:14:48 +0200
Subject: [PATCH] Allow media hotkeys

---
 slock.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/slock.c b/slock.c
index c9cdee2..9781aa8 100644
--- a/slock.c
+++ b/slock.c
@@ -17,6 +17,7 @@
 #include <X11/keysym.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
+#include <X11/XF86keysym.h>
 
 #if HAVE_BSD_AUTH
 #include <login_cap.h>
@@ -122,6 +123,7 @@ readpw(Display *dpy, const char *pws)
 	char buf[32], passwd[256];
 	int num, screen;
 	unsigned int len, color;
+	Window win;
 	KeySym ksym;
 	XEvent ev;
 	static int oldc = INIT;
@@ -170,6 +172,15 @@ readpw(Display *dpy, const char *pws)
 				if (len)
 					--len;
 				break;
+			case XF86XK_AudioLowerVolume:
+			case XF86XK_AudioMute:
+			case XF86XK_AudioRaiseVolume:
+			case XF86XK_AudioPlay:
+			case XF86XK_AudioStop:
+			case XF86XK_AudioPrev:
+			case XF86XK_AudioNext:
+				XSendEvent(dpy, win, True, KeyPressMask, &ev);
+				break;
 			default:
 				if (num && !iscntrl((int)buf[0]) && (len + num < sizeof(passwd))) {
 					memcpy(passwd + len, buf, num);
-- 
2.8.3

