diff --git config.def.h config.def.h
index ce5b9b7..6d720e9 100644
--- config.def.h
+++ config.def.h
@@ -4,9 +4,6 @@
 /* Port used when "-p" is not given */
 #define DEFAULT_PORT "6667"
 
-/* Nickname when "-n" is not given */
-#define NICK "maniac"
-
 /* Timestamp format; see strftime(3). */
 #define TIMESTAMP_FORMAT "%Y-%m-%d %R"
 
diff --git sic.c sic.c
index 72d6026..ecefaf2 100644
--- sic.c
+++ sic.c
@@ -146,7 +146,7 @@ usage(void) {
 int
 main(int argc, char *argv[]) {
 	struct timeval tv;
-	const char *user = NICK;
+	const char *user = getenv("USER");
 	int n;
 	fd_set rd;
 
