diff --git a/config.def.h b/config.def.h
index 1eb9566..fac7fa7 100644
--- a/config.def.h
+++ b/config.def.h
@@ -63,6 +63,18 @@ static Bool allowgeolocation      = TRUE;
 	} \
 }
 
+#define ONLOAD(u) { \
+	.v = (char *[]){"/bin/sh", "-c", \
+		"~/.surf/omnibar addhist \"$0\"", u, NULL \
+	}  \
+}
+
+#define GOTO { \
+	.v = (char *[]){"/bin/sh", "-c", \
+		"~/.surf/omnibar goto $0 $1", winid, "_SURF_GO", NULL \
+	} \
+}
+
 /* styles */
 /*
  * The iteration will stop at the first match, beginning at the beginning of
@@ -110,7 +122,7 @@ static Key keys[] = {
     { MODKEY,               GDK_o,      source,     { 0 } },
     { MODKEY|GDK_SHIFT_MASK,GDK_o,      inspector,  { 0 } },
 
-    { MODKEY,               GDK_g,      spawn,      SETPROP("_SURF_URI", "_SURF_GO") },
+    { MODKEY,               GDK_g,      spawn,      GOTO },
     { MODKEY,               GDK_f,      spawn,      SETPROP("_SURF_FIND", "_SURF_FIND") },
     { MODKEY,               GDK_slash,  spawn,      SETPROP("_SURF_FIND", "_SURF_FIND") },
 
diff --git a/surf.c b/surf.c
index 02656ec..1d84358 100644
--- a/surf.c
+++ b/surf.c
@@ -731,6 +731,8 @@ loadstatuschange(WebKitWebView *view, GParamSpec *pspec, Client *c) {
 			g_object_set(G_OBJECT(set), "user-stylesheet-uri",
 					getstyle(uri), NULL);
 		}
+		Arg a = (Arg)ONLOAD((char *)uri);
+		spawn(NULL, &a);
 		break;
 	case WEBKIT_LOAD_FINISHED:
 		c->progress = 100;
