diff -wu surf-0.6/config.def.h surf-0.6-edit/config.def.h
--- surf-0.6/config.def.h	2013-02-10 19:40:14.000000000 +0100
+++ surf-0.6-edit/config.def.h	2015-11-07 13:12:23.854017366 +0100
@@ -43,6 +43,18 @@
 	} \
 }
 
+#define ONLOAD(u) { \
+	.v = (char *[]){"/bin/sh", "-c", \
+		"~/.surf/omnibar addhist \"$0\"", u, NULL \
+	} \
+}
+
+#define GOTO { \
+	.v = (char *[]){"/bin/sh", "-c", \
+		"~/.surf/omnibar goto \"$0\" \"$1\"", winid, "_SURF_GO", NULL \
+	} \
+}
+
 #define MODKEY GDK_CONTROL_MASK
 
 /* hotkeys */
@@ -80,7 +92,7 @@
     { MODKEY,               GDK_o,      source,     { 0 } },
     { MODKEY|GDK_SHIFT_MASK,GDK_o,      inspector,  { 0 } },
 
-    { MODKEY,               GDK_g,      spawn,      SETPROP("_SURF_URI", "_SURF_GO") },
+    { MODKEY,               GDK_g,      spawn,      GOTO },
     { MODKEY,               GDK_f,      spawn,      SETPROP("_SURF_FIND", "_SURF_FIND") },
     { MODKEY,               GDK_slash,  spawn,      SETPROP("_SURF_FIND", "_SURF_FIND") },
 
Solo in surf-0.6-edit/: config.h
Solo in surf-0.6-edit/: surf-0.6-omnibar.diff
diff -wu surf-0.6/surf.c surf-0.6-edit/surf.c
--- surf-0.6/surf.c	2013-02-10 19:40:14.000000000 +0100
+++ surf-0.6-edit/surf.c	2015-11-07 13:13:24.253997068 +0100
@@ -588,11 +588,11 @@
 	WebKitWebDataSource *src;
 	WebKitNetworkRequest *request;
 	SoupMessage *msg;
-	char *uri;
+	char *uri = geturi(c);
+	Arg arg;
 
 	switch(webkit_web_view_get_load_status (c->view)) {
 	case WEBKIT_LOAD_COMMITTED:
-		uri = geturi(c);
 		if(strstr(uri, "https://") == uri) {
 			frame = webkit_web_view_get_main_frame(c->view);
 			src = webkit_web_frame_get_data_source(frame);
@@ -604,6 +604,8 @@
 		setatom(c, AtomUri, uri);
 		break;
 	case WEBKIT_LOAD_FINISHED:
+		arg = (Arg)ONLOAD(uri);
+		spawn(NULL, &arg);
 		c->progress = 100;
 		update(c);
 		break;
