diff -r 53d98940cb04 config.def.h
--- a/config.def.h	Fri Jun 04 11:41:16 2010 +0100
+++ b/config.def.h	Sun Jun 13 22:53:04 2010 +0200
@@ -12,6 +12,7 @@
 static const unsigned int snap      = 32;       /* snap pixel */
 static const Bool showbar           = True;     /* False means no bar */
 static const Bool topbar            = True;     /* False means bottom bar */
+static const Bool viewonttag        = True;     /* Switch view on tag switch */
 
 /* tagging */
 static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
diff -r 53d98940cb04 dwm.c
--- a/dwm.c	Fri Jun 04 11:41:16 2010 +0100
+++ b/dwm.c	Sun Jun 13 22:53:04 2010 +0200
@@ -1613,6 +1613,8 @@
 	if(selmon->sel && arg->ui & TAGMASK) {
 		selmon->sel->tags = arg->ui & TAGMASK;
 		arrange(selmon);
+		if(viewonttag)
+			view(arg);
 	}
 }
 
