--- a/dwm.c	2009-09-17 10:48:57.000000000 +0200
+++ b/dwm.c	2009-09-16 18:54:37.000000000 +0200
@@ -1270,10 +1270,10 @@
 	XWindowChanges wc;
 
 	if(applysizehints(c, &x, &y, &w, &h, interact)) {
-		c->x = wc.x = x;
-		c->y = wc.y = y;
-		c->w = wc.width = w;
-		c->h = wc.height = h;
+		c->x = wc.x = x + gappx;
+		c->y = wc.y = y + gappx;
+		c->w = wc.width = w - 2 * gappx ;
+		c->h = wc.height = h - 2 * gappx ;
 		wc.border_width = c->bw;
 		XConfigureWindow(dpy, c->win, CWX|CWY|CWWidth|CWHeight|CWBorderWidth, &wc);
 		configure(c);

diff -r 72e52c5333ef config.def.h
--- a/config.def.h	Wed Nov 25 13:56:17 2009 +0000
+++ b/config.def.h	Tue Dec 15 18:39:44 2009 +0100
@@ -7,14 +7,15 @@
+static const unsigned int gappx     = 4;           /* gap pixel between windows */
