diff --git a/dwm.c b/dwm.c
index d9443da..28528ed 100644
--- a/dwm.c
+++ b/dwm.c
@@ -176,6 +176,7 @@ static void destroynotify(XEvent *e);
 static void detach(Client *c);
 static void detachstack(Client *c);
 static Monitor *dirtomon(int dir);
+static void dmenuspawn(const Arg *arg);
 static void drawbar(Monitor *m);
 static void drawbars(void);
 static void drawsquare(Bool filled, Bool empty, Bool invert, unsigned long col[ColLast]);
@@ -709,6 +710,17 @@ dirtomon(int dir) {
 }
 
 void
+dmenuspawn(const Arg *arg) {
+	char monstr[2] = "0";
+	const char *dmenucmd[] = { "dmenu_run", "-fn", font, "-m", monstr, "-nb",
+	           normbgcolor, "-nf", normfgcolor, "-sb", selbgcolor, "-sf",
+	           selfgcolor, NULL };
+	Arg a = { .v = dmenucmd };
+	monstr[0] = '0' + selmon->num;
+	spawn(&a);
+}
+
+void
 drawbar(Monitor *m) {
 	int x;
 	unsigned int i, occ = 0, urg = 0;
