diff --git a/config.def.h b/config.def.h
index 693bdbd..cba3754 100644
--- a/config.def.h
+++ b/config.def.h
@@ -21,7 +21,7 @@ static unsigned int actionfps = 30;
 static char termname[] = "st-256color";
 
 static unsigned int tabspaces = 8;
-
+#define WORD_BREAK " "
 
 /* Terminal colors (16 first used in escape sequence) */
 static const char *colorname[] = {
diff --git a/st.c b/st.c
index 8b1fc56..8ed395c 100644
--- a/st.c
+++ b/st.c
@@ -294,6 +294,7 @@ typedef struct {
 } DC;
 
 static void die(const char *, ...);
+static bool is_word_break(char);
 static void draw(void);
 static void redraw(int);
 static void drawregion(int, int, int, int);
@@ -920,12 +921,12 @@ brelease(XEvent *e) {
 				/* double click to select word */
 				sel.bx = sel.ex;
 				while(sel.bx > 0 && term.line[sel.ey][sel.bx-1].state & GLYPH_SET &&
-						term.line[sel.ey][sel.bx-1].c[0] != ' ') {
+						!is_word_break(term.line[sel.ey][sel.bx-1].c[0])) {
 					sel.bx--;
 				}
 				sel.b.x = sel.bx;
 				while(sel.ex < term.col-1 && term.line[sel.ey][sel.ex+1].state & GLYPH_SET &&
-						term.line[sel.ey][sel.ex+1].c[0] != ' ') {
+						!is_word_break(term.line[sel.ey][sel.ex+1].c[0])) {
 					sel.ex++;
 				}
 				sel.e.x = sel.ex;
@@ -974,6 +975,17 @@ die(const char *errstr, ...) {
 	exit(EXIT_FAILURE);
 }
 
+bool
+is_word_break(char c) {
+	static char *word_break = WORD_BREAK;
+	char *s = word_break;
+	while(*s) {
+		if(*s == c) return true;
+		s++;
+	}
+	return false;
+}
+
 void
 execsh(void) {
 	char **args;
