diff -r db4f3f0ef420 st.c
--- a/st.c	Fri Nov 02 20:08:51 2012 +0100
+++ b/st.c	Sat Nov 03 00:19:19 2012 +0100
@@ -2260,8 +2260,10 @@
 	xw.cw = dc.font.width;
 	xw.ch = dc.font.height;
 
+#ifndef NOBOLDFONT
 	FcPatternDel(pattern, FC_WEIGHT);
 	FcPatternAddInteger(pattern, FC_WEIGHT, FC_WEIGHT_BOLD);
+#endif
 	if(xloadfont(&dc.bfont, pattern))
 		die("st: can't open font %s\n", fontstr);
 
@@ -2397,6 +2399,10 @@
 			/* greyscale */
 			fg = &dc.xft_col[base.fg + 4];
 		}
+#ifdef DEFAULTBOLD
+		if(base.fg == defaultfg)
+			fg = &dc.xft_col[DEFAULTBOLD];
+#endif
 		/*
 		 * Those ranges will not be brightened:
 		 *	8 - 15 – bright system colors
@@ -2406,10 +2412,22 @@
 		font = &dc.bfont;
 	}
 
-	if(base.mode & ATTR_ITALIC)
-		font = &dc.ifont;
-	if(base.mode & (ATTR_ITALIC|ATTR_ITALIC))
-		font = &dc.ibfont;
+	if(base.mode & ATTR_ITALIC) {
+		if(base.mode & ATTR_BOLD) {
+			font = &dc.ibfont;
+		} else {
+			font = &dc.ifont;
+#ifdef DEFAULTITALIC
+			if(base.fg == defaultfg)
+				fg = &dc.xft_col[DEFAULTITALIC];
+#endif
+		}
+	}
+
+#ifdef DEFAULTUNDERLINE
+	if((base.mode & ATTR_UNDERLINE) && base.fg == defaultfg)
+		fg = &dc.xft_col[DEFAULTUNDERLINE];
+#endif
 
 	if(IS_SET(MODE_REVERSE)) {
 		if(fg == &dc.xft_col[defaultfg]) {
