From ef3d847677f5d8675c630fcea6c756881ca20659 Mon Sep 17 00:00:00 2001
From: thatlittlegit <personal@thatlittlegit.tk>
Date: Sat, 28 Apr 2018 16:51:49 -0400
Subject: [PATCH] Don't quit if not all windows are empty

---
 dwm.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/dwm.c b/dwm.c
index c98678d..57043f2 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1248,7 +1248,16 @@ propertynotify(XEvent *e)
 void
 quit(const Arg *arg)
 {
-	running = 0;
+	unsigned int n;
+	Window *junk = malloc(1);
+
+	XQueryTree(dpy, root, junk, junk, &junk, &n);
+
+	if (n == 2) {
+		running = 0;
+	}
+
+	free(junk);
 }
 
 Monitor *
-- 
2.17.0

